aws-api/src/main/java/com/amplifyframework/api/aws/MutiAuthSubscriptionOperation.java

```diff
@@ -151,6 +164,16 @@ public synchronized void cancel() {
         }
     }
 
+    private boolean hasAuthRelatedErrors(GraphQLResponse<T> response) {
+        for (GraphQLResponse.Error error : response.getErrors()) {
+            if (error.getExtensions() != null &&
+                UNAUTHORIZED_EXCEPTION.equals(error.getExtensions().get("errorType"))) {
+                return true;
+            }
```

Comment:
Minor suggestion for improvement that is a little bit more type safe, not necessarily needed for this PR: 

```suggestion
                if (!Empty.check(error.getExtensions())) {
                    AppSyncExtensions extensions = new AppSyncExtensions(error.getExtensions());
                    return AppSyncErrorType.UNAUTHORIZED.equals(extensions.getErrorType());
                }
```

Note this will change will require moving AppSyncExtensions from `aws-datastore` to `aws-api-appsync` first though.

Reply:
I'm testing that. If AppSyncExtensions doesn't have any other baggage, it should be an easy enough move.

Reply:
Just checked this one in. Moving that file didn't cause any side effects